docs/Documentation/Development/Libraries/hsl.md

```diff
@@ -0,0 +1,167 @@
+---
+title: HSL
+postdate: October 27, 2020
+layout: default
+author: Jonathan Maack
+description: How to install and use HSL with with IPOPT
+parent: Libraries
+grand_parent: Development
+---
+
+# HSL for Ipopt
+
+*HSL (Harwell Subroutine Library) for Ipopt are a set of linear solvers that can greatly accelerate the speed of the optimization over the default MUMPS solver.*
+
+## Installation
+
+Go to the [HSL for Ipopt](http://www.hsl.rl.ac.uk/ipopt/) site and follow the instructions to request the source code for all the available solvers. Note that the solver MA27 is free to obtain, but MA27 is a serial solver. Other solvers will require a license. Please request a license that applies to your use case.
+
+### Eagle
+
+The default version of Ipopt distributed with `Ipopt.jl` on Linux links to the OpenBLAS library.  This causes issues when linking the HSL library to the MKL libraries.  For this reason, to use HSL linear solvers with Ipopt on Eagle, either we must compile Ipopt from scratch or compile HSL with OpenBLAS instead of MKL.  For performance reasons, we have elected to compile Ipopt from scratch so that we can use the MKL libraries.
+
+The following provides detailed instructions for compiling Ipopt with HSL and Mumps on Eagle.
+
+#### Pre-requisites
+
+##### Metis
+
+Metis helps the HSL solvers perform better.  Therefore, it is recommended that you also install or build the Metis library.  If you do want to install Metis, it must be done before compiling the HSL library.
+
+On Eagle, the easiest way to install Metis is to use anaconda:
+
+```bash
+module load conda
+conda create -n <conda_environment>
+conda activate <conda_environment>
+conda install -c conda-forge metis
+```
+**Note**: `module load conda` loads the default anaconda module. You may use a different conda module based on your needs.
+
+**Note**: Anaconda packages sometimes have issues when they come from different channels.  We tend to pull everything from `conda-forge` hence the channel choice above.
```

Comment:
I think adding the admonition blocks for the notes and warnings throughout the page would help with readability.